api-publishing/src/main/java/org/zalando/nakadi/PublishingResultConverter.java

```diff
@@ -60,17 +60,22 @@ private EventPublishingStatus mapPublishingStatus(final Check.Status status) {
 
     public EventPublishResult mapPublishingResultToView(final List<NakadiRecordResult> recordsMetadata) {
         final List<BatchItemResponse> batchItemResponses = new LinkedList<>();
-        EventPublishingStatus status = null;
         for (final NakadiRecordResult recordMetadata : recordsMetadata) {
-            status = mapPublishingStatus(recordMetadata.getStatus());
+            final EventPublishingStatus status = mapPublishingStatus(recordMetadata.getStatus());
             batchItemResponses.add(new BatchItemResponse()
                     .setStep(EventPublishingStep.PUBLISHING)
-                    .setPublishingStatus(mapPublishingStatus(recordMetadata.getStatus()))
+                    .setPublishingStatus(status)
                     .setEid(recordMetadata.getMetadata().getEid())
-                    .setDetail(recordMetadata.getException().getMessage()));
+                    .setDetail((recordMetadata.getException() != null) ?
+                            recordMetadata.getException().getMessage() : ""));
         }
 
-        return new EventPublishResult(status, EventPublishingStep.PUBLISHING, batchItemResponses);
+        final var overallStatus = batchItemResponses.stream()
+                .map(BatchItemResponse::getPublishingStatus)
+                .filter(status -> !EventPublishingStatus.SUBMITTED.equals(status))
+                .findAny().map(status -> EventPublishingStatus.FAILED).orElse(EventPublishingStatus.SUBMITTED);
```

Comment:
the publishing status is also used in the controller to prepare usage statistic.

Reply:
so to conclude:
- if there is a failed status in the batch, the publishing result status should be failed
- if there is no failed status, but only aborted the status should be aborted
- otherwise succeeded 

does it make sense ?

Reply:
Makes sense, I'll update it.

Reply:
just thought about it, since we separated validation (checks) and publishing, probably your code make sense and works as expected, but I would just fix it for common understanding and logic.

Reply:
Updated